optuna/samplers/_nsgaiii.py

```diff
@@ -0,0 +1,637 @@
+    in elite population and adds one of borderline front member who has the same closest reference
+    point.
+
+    Args:
+        target_population_size:
+            The number of individuals to select.
+        elite_population_num:
+            The number of individuals which are already selected as the elite population.
+        population:
+            List of all the trials in the current surviving generation.
+        distance_reference_points:
+            A ``numpy.ndarray`` with rows of generation size. Each row is the distance from the
+            corresponding individual to the closest reference point.
+        closest_reference_points:
+            A ``numpy.ndarray`` with rows of generation size. Each row is the index of the closest
+            reference point to the corresponding individual.
+        rng:
+            Random number generator.
+
+    Returns:
+        A list of trials which are selected as the next generation.
+    """
+    if len(population) < target_population_size:
+        raise IndexError(
+            "The population size must be greater than or equal to the target population size."
+        )
+
+    # reference_point_to_borderline_population keeps pairs of a neighbor and the distance of
+    # each reference point from borderline front population.
+    reference_point_to_borderline_population = defaultdict(list)
+    for i, reference_point_idx in enumerate(closest_reference_points[elite_population_num:]):
+        population_idx = i + elite_population_num
+        reference_point_to_borderline_population[reference_point_idx].append(
+            (distance_reference_points[population_idx], i)
+        )
+
+    # reference_points_to_elite_population_count keeps how many elite neighbors each reference
+    # point has.
+    reference_point_to_elite_population_count: dict[int, int] = defaultdict(int)
+    for i, reference_point_idx in enumerate(closest_reference_points[:elite_population_num]):
+        reference_point_to_elite_population_count[reference_point_idx] += 1
+    # nearest_points_count_to_reference_points classifies reference points which have at least one
+    # closest borderline population member by the number of elite neighbors they have.  Each key
+    # corresponds to the number of elite neighbors and the value to the reference point indices.
+    nearest_points_count_to_reference_points = defaultdict(list)
+    for reference_point_idx in reference_point_to_borderline_population:
+        elite_population_count = reference_point_to_elite_population_count[reference_point_idx]
+        nearest_points_count_to_reference_points[elite_population_count].append(
+            reference_point_idx
+        )
+
+    count = -1
+    additional_elite_population: list[FrozenTrial] = []
+    is_shuffled: defaultdict[int, bool] = defaultdict(bool)
+    while len(additional_elite_population) < target_population_size:
+        if not nearest_points_count_to_reference_points[count]:
```

Comment:
`len([]) == 0` might be easier for readers than `not []`.

```suggestion
        if len(nearest_points_count_to_reference_points[count]) == 0:
```

Reply:
I'm sorry, but I got confused. It should be `if len(nearest_points_count_to_reference_points[count]) > 0:`.